mitodl/ocw-studio
# update content dispositions

#### Pre-Flight checklist

#### What are the relevant tickets?
closes https://github.com/mitodl/ocw-studio/issues/1344
#### What's this PR do?
This pr sets content-dispsition to download for pdf files and videos so that the "download" button downloads the file instead of opening it in a browser.

#### How should this be manually tested?
- Use the RC settings for `AWS_ACCOUNT_ID`
- Place 1 small video file in your gdrive `videos_final` folder for a site and then sync.
- Wait 5 minutes or so to make sure the transcode job finished.
- Using Postman or curl, make a request to http://localhost:8043/api/transcode-jobs/ with the following data, and headers `"Authorization: Bearer <settings.API_BEARER_TOKEN>"` and `"Content-Type: application/json"`:
```
{
  "version": "0",
  "id": "c120fe11-87db-c292-b3e5-1cc90740f6e1",
  "detail-type": "MediaConvert Job State Change",
  "source": "aws.mediaconvert",
  "account": "<AWS_ACCOUNT_ID>",
  "time": "2021-08-05T16:52:33Z",
  "region": "us-east-1",
  "resources": [
    "arn:aws:mediaconvert:us-east-1:<AWS_ACCOUNT_ID>:jobs/<VIDEOJOB_ID>"
  ],
  "detail": {
    "timestamp": 1628172900136,
    "accountId": "<AWS_ACCOUNT_ID>",
    "queue": "arn:aws:mediaconvert:us-east-1:<AWS_ACCOUNT_ID>:queues/Default",
    "jobId": "<VIDEOJOB_ID>",
    "status": "COMPLETE",
    "userMetadata": {},
    "outputGroupDetails": [
      {
        "outputDetails": [
          {
            "outputFilePaths": [
              "s3://<AWS_STORAGE_BUCKET_NAME>/aws_mediaconvert_transcodes/<SITE_NAME>/<DRIVE_FILE_ID>/<VIDEO_FILENAME>_360p_16_9.mp4"
            ],
            "durationInMs": 132033,
            "videoDetails": {
              "widthInPx": 640,
              "heightInPx": 360
            }
          },
          {
                "outputDetails": [
                  {
                    "outputFilePaths": [
                      "s3://<AWS_STORAGE_BUCKET_NAME>/aws_mediaconvert_transcodes/<SITE_NAME>/<DRIVE_FILE_ID>/<VIDEO_FILENAME>_youtube.mp4"
                    ],
                    "durationInMs": 132033,
                    "videoDetails": {
                      "widthInPx": 1280,
                      "heightInPx": 720
                    }
                  },
                ],
                "type": "FILE_GROUP"
              }
    ]
  }
}
```

- Go to https://ol-ocw-studio-app-qa.s3.amazonaws.com/courses/<SITE_NAME>/<FILE_NAME>_360p_16_9.mp4 and verify that the file is downloaded instead of opening in the browser

- From the command line, set the destination_id for the VideoFile object to GfjkPmTrhwE, an already transcripted file. Also update the youtube id for the resource to GfjkPmTrhwE

-Publish the course from the ui. Don't worry about the publishing backend being set up, we just need to trigger the pre-publish action. 

-Go to https://ol-ocw-studio-app-qa.s3.amazonaws.com/courses/<SITE_NAME>/<DRIVE_FILE_ID>_transcript.pdf and verify that the file is downloaded instead of opening in the browser
